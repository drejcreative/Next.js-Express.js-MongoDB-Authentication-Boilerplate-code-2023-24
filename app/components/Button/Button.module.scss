.button {
  padding: 11px 20px;
  border-radius: 15px;
  border: 1px solid var(--border);
  color: var(--border);
  display: inline-block;
  cursor: pointer;
  display: inline-flex;
  align-items: center;

  // Loader
  div {
    margin-right: 5px;
  }

  &.primary {
    background: var(--main);
    border: 1px solid var(--main);
    color: var(--light);
    &.thin {
      color: var(--main);
    }
  }
  &.secoundary {
    background: var(--contrast);
    border: 1px solid var(--contrast);
    color: var(--main);
    &.thin {
      color: var(--contrast);
    }
  }
  &.terniary {
    background: var(--focus);
    border: 1px solid var(--focus);
    color: var(--light);
    &.thin {
      color: var(--focus);
    }
  }
  &.error {
    background: var(--error);
    border: 1px solid var(--error);
    color: var(--light);
    &.thin {
      color: var(--error);
    }
  }
  &.disabled {
    background: var(--border);
    border: 1px solid var(--border);
    color: var(--disabled);
    cursor: not-allowed;
    &.thin {
      color: var(--border);
      background: rgba(164, 190, 195, 0.4);
    }
  }
  &.thin {
    background: none;
  }

  &.small {
    padding: 4px 8px;
    border-radius: 4px;
    height: 30px;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .icon {
    margin-right: 10px;
  }
}
