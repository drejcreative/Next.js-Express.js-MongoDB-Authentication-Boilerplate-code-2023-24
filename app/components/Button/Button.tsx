import React from "react";
import { motion } from "framer-motion";
import { Loader } from "../Loader/Loader";

import style from "./Button.module.scss";

interface Props {
  text: string;
  onClick?: () => void;
  primary?: boolean;
  secoundary?: boolean;
  terniary?: boolean;
  error?: boolean;
  thin?: boolean;
  small?: boolean;
  disabled?: boolean;
  loading?: boolean;
  icon?: React.ReactNode;
  inlineStyle?: React.CSSProperties;
}

const Button = ({
  text,
  primary,
  secoundary,
  terniary,
  error,
  thin,
  small,
  disabled,
  inlineStyle,
  loading,
  onClick,
  icon,
}: Props) => {
  return (
    <motion.button
      whileHover={{ scale: 1.05 }}
      whileTap={{ scale: 0.9 }}
      className={` ${style.button} ${primary ? style.primary : ""} ${
        secoundary ? style.secoundary : ""
      }
        ${terniary ? style.terniary : ""} ${error ? style.error : ""}
        ${disabled ? style.disabled : ""} ${thin ? style.thin : ""} ${small ? style.small : ""} `}
      style={inlineStyle}
      type="submit"
      // disabled={disabled ? 'disabled' : ''}
      onClick={disabled || loading ? () => {} : onClick}
    >
      {loading && <Loader small />}
      {!!icon && <span className={style.icon}>{icon}</span>}
      {text}
    </motion.button>
  );
};

export default Button;
