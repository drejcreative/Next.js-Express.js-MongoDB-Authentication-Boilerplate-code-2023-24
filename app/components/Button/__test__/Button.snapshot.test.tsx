import React from "react";
import { render } from "@testing-library/react";

import Button from "../Button";

it("renders Button", () => {
  const { container } = render(<Button text="test" />);
  expect(container).toMatchSnapshot();
});

it("renders Button Primary", () => {
  const { container } = render(<Button text="test" primary />);
  expect(container).toMatchSnapshot();
});

it("renders loader", () => {
  const { container } = render(<Button text="test" loading />);
  expect(container).toMatchSnapshot();
});

it("renders icon", () => {
  const { container } = render(<Button text="test" icon={<span>Icon</span>} />);
  expect(container).toMatchSnapshot();
});
