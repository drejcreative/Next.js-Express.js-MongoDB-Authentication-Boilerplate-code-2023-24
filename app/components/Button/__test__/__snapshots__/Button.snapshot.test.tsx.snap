// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button 1`] = `
<div>
  <button
    class=" button  
         
           "
    type="submit"
  >
    test
  </button>
</div>
`;

exports[`renders Button Primary 1`] = `
<div>
  <button
    class=" button primary 
         
           "
    type="submit"
  >
    test
  </button>
</div>
`;

exports[`renders icon 1`] = `
<div>
  <button
    class=" button  
         
           "
    type="submit"
  >
    <span
      class="icon"
    >
      <span>
        Icon
      </span>
    </span>
    test
  </button>
</div>
`;

exports[`renders loader 1`] = `
<div>
  <button
    class=" button  
         
           "
    type="submit"
  >
    <div
      class="loader small "
    />
    test
  </button>
</div>
`;
